core/src/main/java/org/elasticsearch/action/main/MainResponse.java

```diff
@@ -137,4 +138,26 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public static MainResponse fromXContent(XContentParser parser) {
         return PARSER.apply(parser, null);
     }
+
+    @Override
```

Comment:
Opened https://github.com/elastic/elasticsearch/pull/23352 and will rebase once that is in.